hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-1.x-replication/src/test/java/com/google/cloud/bigtable/hbase1_x/replication/HbaseToCloudBigtableReplicationEndpointDryRunTest.java

```diff
@@ -0,0 +1,161 @@
+  private static Configuration hbaseConfig;
+  private static ReplicationAdmin replicationAdmin;
+
+  @ClassRule
+  public static final BigtableEmulatorRule bigtableEmulator = BigtableEmulatorRule.create();
+
+  private static Connection cbtConnection;
+  private static Connection hbaseConnection;
+
+  private Table hbaseTable;
+  private Table cbtTable;
+
+  @BeforeClass
+  public static void setUpCluster() throws Exception {
+    // Prepare HBase mini cluster configuration
+    Configuration conf = hbaseTestingUtil.getConfiguration();
+    conf.setBoolean(HConstants.REPLICATION_ENABLE_KEY, true);
+    conf.setBoolean(ServerRegionReplicaUtil.REGION_REPLICA_REPLICATION_CONF_KEY, true);
+    conf.setInt(HConstants.HBASE_CLIENT_RETRIES_NUMBER, 5); // less number of retries is needed
+
+    // Set CBT related configs.
+    conf.set("google.bigtable.instance.id", "test-instance");
+    conf.set("google.bigtable.project.id", "test-project");
+    // This config will connect Replication endpoint to the emulator and not the prod CBT.
+    conf.set("google.bigtable.emulator.endpoint.host", "localhost:" + bigtableEmulator.getPort());
+    conf.setBoolean(HBaseToCloudBigtableReplicationConfiguration.ENABLE_DRY_RUN_MODE_KEY, true);
+
+    hbaseTestingUtil.startMiniCluster(2);
+    hbaseConfig = conf;
+    hbaseConfig.setLong(RpcServer.MAX_REQUEST_SIZE, 102400);
+    replicationAdmin = new ReplicationAdmin(hbaseTestingUtil.getConfiguration());
+
+    cbtConnection = BigtableConfiguration.connect(conf);
+    hbaseConnection = hbaseTestingUtil.getConnection();
+
+    // Setup Replication in HBase mini cluster
+    ReplicationPeerConfig peerConfig = new ReplicationPeerConfig();
+    peerConfig.setReplicationEndpointImpl(
+        HbaseToCloudBigtableReplicationEndpoint.class.getTypeName());
+    // Cluster key is required, we don't really have a clusterKey for CBT.
+    peerConfig.setClusterKey(hbaseTestingUtil.getClusterKey());
+    replicationAdmin.addPeer("cbt", peerConfig);
+
+    LOG.info("#################### SETUP COMPLETE ##############################");
+  }
+
+  @Before
+  public void setupTestCase() throws IOException {
+    // Create and set the empty tables
+    TableName table1 = TableName.valueOf(UUID.randomUUID().toString());
+    TableName table2 = TableName.valueOf(UUID.randomUUID().toString());
+    createTables(table1);
+    createTables(table2);
+
+    cbtTable = cbtConnection.getTable(table1);
+    hbaseTable = hbaseConnection.getTable(table1);
+  }
+
+  private void createTables(TableName tableName) throws IOException {
+    // Create table in HBase
+    HTableDescriptor htd = hbaseTestingUtil.createTableDescriptor(tableName.getNameAsString());
+    HColumnDescriptor cf1 = new HColumnDescriptor(TestUtils.CF1);
+    cf1.setMaxVersions(100);
+    htd.addFamily(cf1);
+    HColumnDescriptor cf2 = new HColumnDescriptor(TestUtils.CF2);
+    cf2.setMaxVersions(100);
+    htd.addFamily(cf2);
```

Comment:
why multiple families? why bother setting maxVersions? It just adds noise

Reply:
By default hbase supports only 1 version and BT supports more. If we don't set this tests that insert more than 1 cell fail. This is to prevent that scenario. The other tests in the package do insert more than 1 cell/column. I will remove multiple families, we don't need them for the dry-runs.

Reply:
It is not required for this test, so I will remove it here.